Add some unit tests from cm3leon involving repeat_interleave

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102733

These actually were fixed by https://github.com/pytorch/pytorch/pull/102570
but that PR doesn't test guard-freeness, so here you go.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

ezyang committed:
Add some unit tests from cm3leon involving repeat_interleave

These actually were fixed by https://github.com/pytorch/pytorch/pull/102570
but that PR doesn't test guard-freeness, so here you go.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

[ghstack-poisoned]

peterbell10 commented on test/test_proxy_tensor.py:
```diff
@@ -929,6 +929,35 @@ def f(x):
 
+    def test_repeat_interleave(self):
+        def f(src_tokens, beam_size):
+            return src_tokens.repeat_interleave(beam_size, 0)
```
Should we have a test where `beam_size` is symbolic?

ezyang replied:
Sure I can make this symbolic too.

peterbell10 commented on test/test_proxy_tensor.py:
```diff
@@ -929,6 +929,35 @@ def f(x):
+        batch_size = 4
+        src_tokens = torch.randint(1, vocab_size, (batch_size, prompt_size))
+        gm = make_fx(f, tracing_mode="symbolic")(src_tokens, 1)
+        self.assertEqual(len(gm.shape_env.guards), 0)
```
Cool, didn't know it was this easy to inspect the guards.

ezyang committed:
Update on "Add some unit tests from cm3leon involving repeat_interleave"


These actually were fixed by https://github.com/pytorch/pytorch/pull/102570
but that PR doesn't test guard-freeness, so here you go.

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge
